components/engine_rocks_helper/src/sst_recovery.rs

```diff
@@ -0,0 +1,256 @@
+        check_duration: Duration,
+    ) -> Self {
+        RecoveryRunner {
+            db,
+            store_meta,
+            damaged_files: vec![],
+            max_hang_duration,
+            check_duration,
+        }
+    }
+
+    fn generate_scheduling_tasks(&mut self, path: &str) {
+        if self.exist_scheduling_regions(path) {
+            return;
+        }
+
+        let live_files = self.db.get_live_files();
+        for i in 0..live_files.get_files_count() {
+            if path == live_files.get_name(i as i32) {
+                let f = FileInfo {
+                    name: live_files.get_name(i as i32),
+                    smallest_key: live_files.get_smallestkey(i as i32).to_owned(),
+                    largest_key: live_files.get_largestkey(i as i32).to_owned(),
+                    overlap_region_ids: Vec::new(),
+                    start_time: Instant::now(),
+                };
+
+                // only support recovering data key for now.
+                if !keys::validate_data_key(&f.smallest_key)
+                    || (!keys::validate_data_key(&f.largest_key)
+                        && f.largest_key.as_slice() != keys::DATA_MAX_KEY)
+                {
+                    set_panic_mark_and_panic(
+                        path,
+                        &format!(
+                            "key range mismatch, smallest key:{:?}, largest key:{:?}",
+                            &f.smallest_key, &f.largest_key
+                        ),
+                    );
+                }
+
+                // lock the store_meta and check if the range exist.
+                if !self.check_overlap_damaged_regions(f) {
+                    set_panic_mark_and_panic(path, "Can't find overlap regions");
+                }
+
+                return;
+            }
+        }
+    }
+
+    // `sst_path` has been processed and is still in a damaged state.
+    fn exist_scheduling_regions(&self, sst_path: &str) -> bool {
+        self.damaged_files.iter().any(|f| f.name == sst_path)
+    }
+
+    // Periodically check for recorded damaged files。
+    //
+    // Acquire meta lock.
+    fn check_damaged_files(&mut self) {
+        let mut new_damaged_files = self.damaged_files.clone();
+        new_damaged_files.retain(|f| {
+            if f.start_time.elapsed() > self.max_hang_duration {
+                set_panic_mark_and_panic(
+                    &f.name,
+                    &format!("recovery job exceeded {:?}", self.max_hang_duration),
+                );
+            }
+            let need_retain = self.check_overlap_damaged_regions(f.clone());
+            if !need_retain {
+                println!(
+                    "ready delete_file {} {:?} {:?} (this line must be deleted before merge PR)",
+                    f.name, &f.smallest_key, &f.largest_key
+                );
+                // set `include_end` to `true` to ensure that sst with the same start key and end key.
+                self.db
```

Comment:
Add some metrics for the check?

Reply:
what check?

Reply:
`check_damaged_files`

Reply:
only add guage to stat damaged files, what do you mean here? add check ops?